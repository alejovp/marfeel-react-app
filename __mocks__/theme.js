export const themeMock = {
    tipography: {
        primaryFontColor: 'black',
        fontFamily: 'Helvetica, sans-serif'
    },
    header: {
        padding: 15
    },
    sectionMenu: {
        color: 'orange',
        textSize: 1.1 //em
    },
    background: {
        color: 'gray'
    },
    burgerButton: {
        color: 'red'
    }
};
