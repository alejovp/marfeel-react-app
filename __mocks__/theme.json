{
    "tipography": {
        "primaryFontColor": "black",
        "fontFamily": "Helvetica, sans-serif"
    },
    "header": {
        "padding": 15
    },
    "sectionMenu": {
        "color": "orange",
        "textSize": 1.1
    },
    "background": {
        "color": "gray"
    },
    "burgerButton": {
        "color": "red"
    },
    "logo": {
        "image": "https://www.marfeel.com/public/images/landing/header/logo-marfeel.svg",
        "size": 36
    }
}