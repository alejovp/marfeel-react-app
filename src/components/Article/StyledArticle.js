import styled from 'styled-components';
import { device } from '../../styles/device';


export const StyledArticle = styled.div`
    display: flex;
    align-items: center;
    justify-content: center;
    width: 100%;
    padding: 20px;
    flex-wrap: wrap;

    @media ${device.tablet} {
        flex-wrap: nowrap;
        align-items: unset;
    }
`;

StyledArticle.displayName = 'StyledArticle';
