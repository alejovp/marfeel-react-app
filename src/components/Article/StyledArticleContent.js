import styled from 'styled-components';

export const StyledArticleContent = styled.div`
    display: flex;
    flex-direction: column;
`;

StyledArticleContent.displayName = 'StyledArticleContent';
