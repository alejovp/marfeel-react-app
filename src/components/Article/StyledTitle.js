import styled from 'styled-components';

export const StyledTitle = styled.h3`
    margin-top: 0;
`;

StyledTitle.displayName = 'StyledTitle';
