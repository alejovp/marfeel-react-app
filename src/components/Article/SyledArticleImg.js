import styled from 'styled-components';
import { device } from '../../styles/device';

export const StyledArticleImg = styled.img`
    width: 200px;
    height: 150px;
    object-fit: cover;
    margin-bottom: 20px;

    @media ${device.mobileL} { 
        width: 350px;
        height: 250px;
    }

    @media ${device.tablet} {
        margin-bottom: 0;
        margin-right: 20px;
    }
`;

StyledArticleImg.displayName = 'StyledArticleImg';
