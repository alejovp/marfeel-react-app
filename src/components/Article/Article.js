import React from 'react';
import PropTypes from 'prop-types';
import { StyledArticle } from './StyledArticle';
import { StyledArticleContent } from './StyledArticleContent';
import { StyledTitle } from './StyledTitle';
import { StyledArticleImg } from './SyledArticleImg';


export const Article = ({ img, title, description}) => {

    if (!title || !description) {
        return null;
    }

    const renderImage = () => {
        if (!img) {
            return null;
        }
        return <StyledArticleImg src={img} />;
    };

    return (
        <StyledArticle>
            {renderImage()}
            <StyledArticleContent>
                <StyledTitle>
                    {title}
                </StyledTitle>
                <div>
                    <p>{description}</p>
                </div>
            </StyledArticleContent>
        </StyledArticle>
    );
};

Article.propTypes = {
    img: PropTypes.string,
    title: PropTypes.string,
    description: PropTypes.string
};
