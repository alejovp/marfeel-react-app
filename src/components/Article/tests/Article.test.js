import React from 'react';
import { Article } from '../Article';
import theme from '../../../../__mocks__/theme.json';
import { shallowWithTheme } from '../../../../scripts/jest/setup';
import { StyledArticle } from '../StyledArticle';
import { StyledArticleImg } from '../SyledArticleImg';
import { StyledArticleContent } from '../StyledArticleContent';
import { StyledTitle } from '../StyledTitle';


const articleMock = {
    title: 'Test article',
    description: 'Test description...',
    img: 'test-img-url'
};

describe('<Article />', () => {

    describe('shape', () => {
        const { title, description, img } = articleMock;
        const renderedComponent = shallowWithTheme(
            <Article title={title}
                     description={description}
                     img={img} />,
            theme
        );

        it('will render correctly', () => {
            expect(renderedComponent).toMatchSnapshot();
        });

        it('will render a StyledArticle component', () => {
            expect(renderedComponent.find(StyledArticle)).toHaveLength(1);
        });

        it('will render a StyledArticleImg component', () => {
            expect(renderedComponent.find(StyledArticleImg)).toHaveLength(1);
        });

        it('will render a StyledArticleContent component', () => {
            expect(renderedComponent.find(StyledArticleContent)).toHaveLength(1);
        });

    });

    describe('props', () => {

        it('wont render if title prop is undefined', () => {
            const renderedComponent = shallowWithTheme(
                <Article description={articleMock.description}
                         img={articleMock.img} />,
                theme
            );
            expect(renderedComponent.type()).toBe(null);
        });

        it('wont render if description prop is undefined', () => {
            const renderedComponent = shallowWithTheme(
                <Article title={articleMock.title}
                         img={articleMock.img} />,
                theme
            );
            expect(renderedComponent.type()).toBe(null);
        });

        it('wont render StyledArticleImg if img prop is undefined', () => {
            const renderedComponent = shallowWithTheme(
                <Article title={articleMock.title}
                         description={articleMock.description} />,
                theme
            );
            expect(renderedComponent.find(StyledArticleImg)).toHaveLength(0);
        });

        const renderedComponent = shallowWithTheme(
            <Article title={articleMock.title}
                     description={articleMock.description}
                     img={articleMock.img} />,
            theme
        );

        it('will pass the img prop to the src attr of StyledArticleImg', () => {
            const styledImgComp = renderedComponent.find(StyledArticleImg);
            expect(styledImgComp.props().src).toBe(articleMock.img);
        });

        it('will pass the title prop as the StyledTitle children prop', () => {
            const styledTitle = renderedComponent.find(StyledTitle);
            expect(styledTitle.props().children).toBe(articleMock.title);
        });

        it('will render a <div> with the description prop as text for a <p> children', () => {
            const renderedDiv = renderedComponent.find('div');
            expect(renderedDiv).toHaveLength(1);
            const renderedP = renderedDiv.find('p');
            expect(renderedP.text()).toBe(articleMock.description);
        });

    });

});
