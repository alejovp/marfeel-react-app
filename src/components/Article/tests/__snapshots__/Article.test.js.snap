// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Article /> shape will render correctly 1`] = `
<StyledArticle>
  <StyledArticleImg
    src="test-img-url"
  />
  <StyledArticleContent>
    <StyledTitle>
      Test article
    </StyledTitle>
    <div>
      <p>
        Test description...
      </p>
    </div>
  </StyledArticleContent>
</StyledArticle>
`;
